### PR TITLE
[Feature] Mutation policy

### PR DESCRIPTION
**Pull request by ekoop-terasky**
1. **What this PR does / why we need it**:

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**
Related task:
[TMC-35638](https://jira.eng.vmware.com/browse/TMC-35638)

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by ramya-bangera — APPROVED** [review]

File: internal/resources/policy/kind/mutation/recipe/label_flatten_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+				TargetKubernetesResources: []*policyrecipecustomcommonmodel.VmwareTanzuManageV1alpha1CommonPolicySpecCustomV1TargetKubernetesResources{
+					{
+						APIGroups: []string{"policy"},
+						Kinds:     []string{"pod"},
+					},
+				},
+			},
+			expected: []interface{}{
+				map[string]interface{}{
+					LabelKey: []interface{}{
```
Comment (ausafaq) [inline internal/resources/policy/kind/mutation/recipe/label_flatten_test.go:45]: Is `LabelKey` an array of maps here? Please correct me if I'm wrong.

File: internal/resources/policy/kind/mutation/recipe/label_schema.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			targetKubernetesResourcesKey: common.TargetKubernetesResourcesSchema,
+			scopeKey: {
+				Type:         schema.TypeString,
+				Description:  "Scope",
+				Optional:     true,
+				Default:      "*",
+				ValidateFunc: validation.StringInSlice([]string{"*", "Cluster", "Namespaced"}, false),
+			},
+			LabelKey: {
+				Type:     schema.TypeList,
```
Comment (ausafaq) [inline internal/resources/policy/kind/mutation/recipe/label_schema.go:37]: Can `LabelKey` be a map instead of a list?

**Review by vrabbi — APPROVED** [review]
looks great!
LGTM

File: internal/resources/policy/kind/mutation/recipe/label_schema.go (hunk 2 above)
Comment (ekoop-terasky) [reply internal/resources/policy/kind/mutation/recipe/label_schema.go:37]: I tried at some point but it was not suitable for this case and previously I got following error:
* resource tanzu-mission-control_mutation_policy: label: TypeMap with Elem *Resource not supported,use TypeList/TypeSet with Elem *Resource or TypeMap with Elem *Schema

File: internal/resources/policy/kind/mutation/recipe/label_flatten_test.go (hunk 1 above)
Comment (ekoop-terasky) [reply internal/resources/policy/kind/mutation/recipe/label_flatten_test.go:45]: Almost, slice containing a single map. Quite common approach in this project.

**Review by ausafaq — APPROVED** [review]

**Review by ausafaq — APPROVED** [review]

**Pull request merged**